lib/Fmt_ast.ml

```diff
@@ -884,10 +884,10 @@ and fmt_core_type c ?(box = true) ?pro ?(pro_space = true) ?constraint_ctx
         | _ ->
             list rfs
               ( if
-                  in_type_declaration
-                  && Poly.(c.conf.fmt_opts.type_decl.v = `Sparse)
-                then "@;<1000 0>| "
-                else "@ | " )
+                in_type_declaration
+                && Poly.(c.conf.fmt_opts.type_decl.v = `Sparse)
+              then "@;<1000 0>| "
+              else "@ | " )
```

Comment:
There are some stray whitespace changes here and below.  `make test` on this branch is failing as a result.  Running `dune build @fmt --auto-promote` should fix it.